zebra/if_netlink.c

```diff
@@ -2039,7 +1867,10 @@ int netlink_link_change(struct nlmsghdr *h, ns_id_t ns_id, int startup)
 		if (linkinfo[IFLA_INFO_SLAVE_KIND])
 			slave_kind = RTA_DATA(linkinfo[IFLA_INFO_SLAVE_KIND]);
 
-		netlink_determine_zebra_iftype(kind, &zif_type);
+		if ((slave_kind != NULL) && strcmp(slave_kind, "bond") == 0)
+			netlink_determine_zebra_iftype("bond_slave", &zif_type);
+		else
+			netlink_determine_zebra_iftype(kind, &zif_type);
```

Comment:
I understand that it's just a copy-paste from `netlink_interface`, but I think this change is wrong.
Why should we replace the interface type with generic `bond_slave` when it is actually, for example, `veth` or `gre`?
The fact that the interface is a bond slave is determined by `zif_slave_type` field, look at `IS_ZEBRA_IF_BOND_SLAVE`.

Reply:
Please file an issue so that we can look at this in the future.  I'm knee deep in other stuff and I while I agree that this is not optimal/correct, it's going to operate the same with the other changes at this point in time.  I do agree that this is a problem that should be looked into at some point in the future though

Reply:
#13532

Reply:
This can't be merged as-is because it will break stuff. This code worked only because it was only in `netlink_interface` which is called only on startup. When the interface is moved to a bond after the startup, `netlink_link_change` is called, which doesn't have the wrong code and sets the interface type correctly.

Just look at my PR https://github.com/FRRouting/frr/pull/13545 – the "bond_slave" is never used in the code. You can either merge my PR and rebase yours, or just drop this change. I'm good with both.